hack/tools/go.mod

```
@@ -1,6 +1,6 @@
 module github.com/metal3-io/cluster-api-provider-metal3/hack/tools
 
-go 1.22.8
+go 1.23.0
```

```suggestion
go 1.23
```
Unless specific patch is needed, just list the minor.